Dockerfile

```diff
@@ -143,6 +143,7 @@ ARG PGBIN=/usr/lib/postgresql/$PG_MAJOR/bin
 
 ENV LC_ALL=$LC_ALL LANG=$LANG EDITOR=/usr/bin/editor
 ENV PGDATA=$PGDATA PATH=$PATH:$PGBIN
+ENV DUMB_INIT_SETSID=0
```

Comment:
Currently dump-init is only used for haproxy/confd container.
I am not sure if this is a good thing to have this option for it.

I would suggest instead to introduce dump-init for patroni containers and have this option directly in the entrypoint.sh file: https://github.com/zalando/patroni/blob/552e8643d9d5555ecaf7a882638bcc8302dc6933/docker/entrypoint.sh#L75